### PR TITLE
pastel: init at 0.5.3

### PR DESCRIPTION
**Pull request by davidtwco**
###### Motivation for this change
Add a package for [pastel](https://github.com/sharkdp/pastel).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers
N/A

### CODE REVIEW

File: pkgs/applications/misc/pastel/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  src = fetchFromGitHub {
+    owner = "sharkdp";
+    repo = pname;
+    rev = version;
+    sha256 = "0f54p3pzfp7xrwlqn61l7j41vmgcfph3bhq2khxh5apfwwdx9nng";
+  };
+
+  cargoSha256 = "05yvlm7z3zfn8qd8nb9zpch9xsfzidrpyrgg2vij3h3q095mdm66";
+
+  nativeBuildInputs = [];
```
Comment (alexarice) [inline pkgs/applications/misc/pastel/default.nix]: This is unneccesary

File: pkgs/applications/misc/pastel/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{ stdenv, fetchFromGitHub, rustPlatform, Security }:
```
Comment (alexarice) [inline pkgs/applications/misc/pastel/default.nix]: This is failing due to Security not being input on nixos. You should look how other packages do this and copy them

File: pkgs/applications/misc/pastel/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+rustPlatform.buildRustPackage rec {
+  pname = "pastel";
+  version = "v0.5.3";
```
Comment (alexarice) [inline pkgs/applications/misc/pastel/default.nix]: Can the version only have the numeric value and then the `v` can be added below

File: pkgs/applications/misc/pastel/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+  src = fetchFromGitHub {
+    owner = "sharkdp";
+    repo = pname;
+    rev = version;
```
Comment (alexarice) [inline pkgs/applications/misc/pastel/default.nix]:
```suggestion
    rev = "v${version}";
```

File: pkgs/applications/misc/pastel/default.nix [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    sha256 = "0f54p3pzfp7xrwlqn61l7j41vmgcfph3bhq2khxh5apfwwdx9nng";
+  };
+
+  cargoSha256 = "05yvlm7z3zfn8qd8nb9zpch9xsfzidrpyrgg2vij3h3q095mdm66";
+
+  nativeBuildInputs = [];
+  buildInputs = (stdenv.lib.optional stdenv.isDarwin Security);
```
Comment (alexarice) [inline pkgs/applications/misc/pastel/default.nix]: brackets are not needed

File: pkgs/top-level/all-packages.nix [hunk 6]
```diff
@@ -1872,6 +1872,8 @@ in
 
   optar = callPackage ../tools/graphics/optar {};
 
+  pastel = callPackage ../applications/misc/pastel { };
```
Comment (jonringer) [inline pkgs/top-level/all-packages.nix]: I'll throw you a bone to fix the eval, :)
```suggestion
  pastel = callPackage ../applications/misc/pastel {
    inherit (darwin.apple_sdk.frameworks) Security;
  };
```

File: pkgs/applications/misc/pastel/default.nix (hunk 1 above)
Comment (davidtwco) [reply pkgs/applications/misc/pastel/default.nix]: Removed, thanks.

File: pkgs/applications/misc/pastel/default.nix (hunk 2 above)
Comment (davidtwco) [reply pkgs/applications/misc/pastel/default.nix]: Fixed.

File: pkgs/applications/misc/pastel/default.nix (hunk 3 above)
Comment (davidtwco) [reply pkgs/applications/misc/pastel/default.nix]: Fixed.

File: pkgs/applications/misc/pastel/default.nix (hunk 5 above)
Comment (davidtwco) [reply pkgs/applications/misc/pastel/default.nix]: Fixed.

File: pkgs/top-level/all-packages.nix (hunk 6 above)
Comment (davidtwco) [reply pkgs/top-level/all-packages.nix]: Thanks! I had done this in my local configuration but it completely slipped my mind when making this PR.

**Review by alexarice — APPROVED** [review]
Builds with nix-review and runs on nixos

**Comment by dtzWill** [comment]
What's the relation to #67960? One of these probably should be closed...? :)

**Comment by alexarice** [comment]
> What's the relation to #67960? One of these probably should be closed...? :)

I've managed to review both somehow

**Comment by mtrsk** [comment]
> What's the relation to #67960? One of these probably should be closed...? :)

I can close mine so this one can be merged.

**Review by worldofpeace — APPROVED** [review]

**Comment by worldofpeace** [comment]
Thanks for contributing @davidtwco.

Sorry about the PR duplication @mtrsk, this can occasionally happen with how active the nixpkgs contributors are with packaging.

**Comment by mtrsk** [comment]
> Sorry about the PR duplication @mtrsk, this can occasionally happen with how active the nixpkgs contributors are with packaging.

No prob, I still plan to help the project, especially with broken packages that appear now and then.

**Pull request merged**